src/librustc_trans/back/write.rs

```diff
@@ -557,9 +561,15 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
 
         if config.emit_obj {
             let path = output_names.with_extension(&format!("{}.o", name_extra));
-            with_codegen(tm, llmod, config.no_builtins, |cpm| {
-                write_output_file(cgcx.handler, tm, cpm, llmod, &path, llvm::ObjectFileType);
-            });
+            if !cgcx.is_like_pnacl {
+                with_codegen(tm, llmod, config.no_builtins, |cpm| {
```

Comment:
It sounds like we may want to take a similar strategy to those tools and instead of having the default output type be an object file for insertion into an archive we just have the default output type be bitcode to insert into an archive?